lua/toggleterm/terminal.lua

```diff
@@ -325,13 +327,31 @@ end
 ---@private
 ---Pass self as first parameter to callback
 function Terminal:__stdout()
-  if self.on_stdout then return function(...) self.on_stdout(self, ...) end end
+  if self.auto_scroll or self.on_stdout then
+    return function(...)
+      if self.auto_scroll then
+        vim.api.nvim_buf_call(self.bufnr, function() vim.cmd('norm G') end)
```

Comment:
This should re-use the exising function for this https://github.com/akinsho/toggleterm.nvim/blob/5ea823359b450bcf9a784f73734fd2d52e9de0ec/lua/toggleterm/ui.lua#L181